### PR TITLE
L2 861 sns proposal topic

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Add SnsDecentralizationSale proposal support to the launchpad

# Changes

- show mock proposals if none available
- sns proposal loading
- SnsProposalsStore
- OpenForVotesSnsProposalsStore
- extend dummy proposals with `SetSnsTokenSwapOpenTimeWindow`

# Tests

Will be provided in a separate PR https://dfinity.atlassian.net/browse/L2-877

### CODE REVIEW

File: frontend/svelte/src/lib/services/proposals.services.ts [hunk 1]
```diff
@@ -161,15 +161,20 @@ const findProposals = async ({
   });
 };
 
-// Simplified demo function because currently it's not clear which proposal type and which store will be used.
 // TODO L2-751: switch to real data
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/services/proposals.services.ts:164]: that todo can be removed?

File: frontend/svelte/src/lib/services/sns.services.ts [hunk 2]
```diff
@@ -163,62 +166,113 @@ const mockAbout5SecondsWaiting = <T>(generator: () => T): Promise<T> =>
...
+
+  return queryAndUpdate<ProposalInfo[], unknown>({
+    request: ({ certified, identity }) =>
+      loadProposalsByTopic({
+        certified,
+        identity,
+        topic: Topic.SnsDecentralizationSale,
+      }),
+    onLoad: async ({ response: proposals, certified }) => {
+      console.log("listSnsProposals", proposals);
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/services/sns.services.ts:180]: ultimately to be removed I guess? // TODO

File: frontend/svelte/src/lib/services/proposals.services.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/services/proposals.services.ts:164]: @peterpeterparker  Should the mock data be removed? (that are shown when no proposal is available)

Comment (peterpeterparker) [reply frontend/svelte/src/lib/services/proposals.services.ts:164]: Good question, what do you think @mstrasinskis ?

File: frontend/svelte/src/lib/components/launchpad/Proposals.svelte [hunk 3]
```diff
@@ -3,20 +3,26 @@
...
   import CardGrid from "../ui/CardGrid.svelte";
   import SkeletonProposalCard from "../ui/SkeletonProposalCard.svelte";
   import ProposalCard from "./ProposalCard.svelte";
 
   let loading: boolean = false;
-  let proposals: ProposalInfo[] | undefined = undefined;
-
-  const load = async () => {
-    loading = true;
+  $: isNullable($snsProposalsStore);
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/components/launchpad/Proposals.svelte:17]: not sure what's the goal of this?

File: frontend/svelte/src/lib/components/launchpad/Proposals.svelte [hunk 4]
```diff
@@ -3,20 +3,26 @@
...
 
-    // TODO L2-751: replace the source
-    proposals = await listSnsProposals();
+  let proposals: ProposalInfo[] | undefined = undefined;
+  $: proposals = $openForVotesSnsProposalsStore;
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/components/launchpad/Proposals.svelte:20]: to some extension, you can probably ditch `proposals` local variable and access `$openForVotesSnsProposalsStore` dirrectly in the template

File: frontend/svelte/src/lib/services/proposals.services.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/services/proposals.services.ts:164]: Was removed 👍

File: frontend/svelte/src/lib/components/launchpad/Proposals.svelte (hunk 3 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/components/launchpad/Proposals.svelte:17]: Good catch! It's a 🐞

**Review by peterpeterparker — APPROVED** [review]
Super, thanks!

**Pull request merged**